### PR TITLE
Jeff cohere/droplet self collection

### PR DESCRIPTION
**Pull request by jeff-cohere**
This is a really simple conversion. With `iparam=3`, the `droplet_self_collection` subroutine just zeros its output.

### CODE REVIEW

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+::droplet_self_collection(const Spack& rho, const Spack& inv_rho,
+                          const Spack& qc_incld, const Spack& mu_c,
+                          const Spack& nu, const Spack& ncautc, Spack& ncslf)
+{
+  constexpr Scalar qsmall = C::QSMALL;
+
+  const auto qc_not_small = (qc_incld >= qsmall);
+  if (qc_not_small.any()) {
+    // Khroutdinov and Kogan (2000)
+    ncslf.set(qc_not_small, 0.0);
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: Use int literal.

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/*
+ * Implementation of p3 droplet self collection function. Clients should NOT
+ * #include this file, but include p3_functions.hpp instead.
+ */
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::droplet_self_collection(const Spack& rho, const Spack& inv_rho,
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: It looks like most of these arguments aren't even used? You should remove the ones that aren't used if we are going to assume iparam=3.

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                          const Spack& nu, const Spack& ncautc, Spack& ncslf)
+{
+  constexpr Scalar qsmall = C::QSMALL;
+
+  const auto qc_not_small = (qc_incld >= qsmall);
+  if (qc_not_small.any()) {
+    // Khroutdinov and Kogan (2000)
+    ncslf.set(qc_not_small, 0.0);
+
+    ncslf.set(ncslf == 0, 0);
```
Comment (jgfouca) [inline components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: I don't think you need this line.

**Comment by jeff-cohere** [comment]
If we're going to eventually add other models, this function is a useful indicator of where this physics goes. But I agree it is a case of overkill at the moment. Thanks for the pointers on our style (omitting arguments, etc).

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp (hunk 2 above)
Comment (jeff-cohere) [reply components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: Is this just a matter of practice, or is it for silencing "unused var" warnings? I could see that making things more difficult for someone who adds another setting to this model (unless we're not going to have them revisit this same function). Some folks use UNUSED() macros to indicate that arguments aren't used, for example.

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp (hunk 3 above)
Comment (jeff-cohere) [reply components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: Should we just replace with this, then?

```
ncslf.set(0);
```

I was just trying to be conservative and do what the Fortran code does. Those two statements technically leave alone entries that are smaller than qsmall but nonzero.

**Comment by jgfouca** [comment]
> If we're going to eventually add other models, this function is a useful indicator of where this physics goes. But I agree it is a case of overkill at the moment. Thanks for the pointers on our style (omitting arguments, etc).

I'm OK keeping it, esp since you've already done the work.

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp (hunk 3 above)
Comment (jgfouca) [reply components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: We want to mimic the fortran behavior, so keep the qc_not_small set.

The second line seems redundant. If ncslf is zero, set ncslf to zero?

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp (hunk 2 above)
Comment (jgfouca) [reply components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: You might be able to use anonymous argument to remove the warnings.

File: components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp (hunk 3 above)
Comment (jeff-cohere) [reply components/scream/src/physics/p3/p3_functions_droplet_self_coll_impl.hpp]: Yeah, I guess I assumed something else was going on. We shouldn't be using == with floating point types in the first place, if we really want to be careful. I'll ax it.

**Comment by jgfouca** [comment]
Looks like you have GPU build problems.

**Comment by jeff-cohere** [comment]
Interesting. I'll investigate.

**Comment by jeff-cohere** [comment]
Sorry for the repeated pinging, @jgfouca . :-) Looks like this is just about finished.

**Pull request merged**